### PR TITLE
feat(button): permit disable toggle for `loading`

### LINKED ISSUE #392: Button loading prop should not disable button
## Request

The `loading` prop on the `Button` component disables the button. I don't believe that this should be the default behavior. Let the consumer decide whether to disable the button or not. If they already pass in a prop to set `loading`, they can use the same one to disable if they want.

Not sure if it is a bug or desired behavior, but when setting `loading` to true, the button changes to a ghost button, regardless of initial type.

### PR DESCRIPTION
**Pull request by SimonFinney**
## Affected issues

Resolves #392

## Proposed changes

- Accommodate toggling both `disabled` and `loading` for buttons simultaneously
- Enforce disabling button for `SummaryCardAction` to maintain existing design

### CODE REVIEW

File: src/components/Button/Button.js [hunk 1]
```diff
@@ -66,9 +63,6 @@ Button.defaultProps = {
 Button.propTypes = {
   ...propTypes,
 
-  /** @type {boolean} Whether or not the button is disabled. */
-  disabled: PropTypes.bool,
-
   /** @type {boolean} Whether or not the button is in a loading state. While loading, the button is disabled & icons provided via the `renderIcon` prop will not be shown. */
```
Comment (jendowns) [inline src/components/Button/Button.js:66]: This prop description will need to be updated 👍

**Comment by SimonFinney** [comment]
@jendowns Good catch! Because we're now allowing consumers to determine whether a loading button should be disabled or not, we likely now need a visual indicator to differentiate between the two variations.

Reverting to Carbon's default behavior for these styles seemed like a reasonable approach, but I'd like to @cameroncalder or @petervachon to chime in if we require additional consideration here.

**Comment by jendowns** [comment]
@SimonFinney The origin designs had the `SummaryCard`'s loading button with the blue60 color text, I believe. For the normal button, I don't necessarily have an opinion about that. But for that `SummaryCard`, it's a design regression. 👍

File: src/components/Button/Button.js (hunk 1 above)
Comment (SimonFinney) [reply src/components/Button/Button.js:66]: What do you mean here?

**Pull request merged**